packages/server/dataloader/customLoaderMakers.ts

```diff
@@ -62,8 +66,17 @@ export const users = () => {
 }
 
 export const serializeUserTasksKey = (key: UserTasksKey) => {
-  const userIdKey = key.userIds ? key.userIds.sort().join(':') : '*'
-  return `${userIdKey}:${key.teamIds.sort().join(':')}:${key.first}:${key.after}:${key.archived}`
+  const {userIds, teamIds, first, after, archived, statusFilters, filterQuery} = key
+  const parts = [
+    isNonEmptyArray(userIds) ? userIds!.sort().join(':') : '*',
```

Comment:
having to manually assert userIds is nonnull makes this function less helpful.
Instead, I'd propose writing `userIds?.length`, which does the same thing, with less typing!

fun fact! before optional chaining (`?.`) you could accomplish this with something called typeguards:

```ts
export default (thing: any): thing is [] => Array.isArray(thing) && thing.length > 0
```

Reply:
that is much more succinct!